### PR TITLE
encrypt

### PR DESCRIPTION
**Pull request by nadezdatsygankova**

### CODE REVIEW

File: backend/package-lock.json [hunk 1]
```diff
@@ -1,2375 +1,8 @@
 {
   "name": "backend",
   "version": "1.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
   "requires": true,
```
Comment (i-iurchuk) [inline backend/package-lock.json:5]: what version of npm and node are you using?)

**Review by i-iurchuk — COMMENTED** [review]
Great first pull request! :+1:

File: backend/package-lock.json (hunk 1 above)
Comment (nadezdatsygankova) [reply backend/package-lock.json:5]: npm: '6.14.15', node: '14.18.1'

Comment (nadezdatsygankova) [reply backend/package-lock.json:5]: > what version of npm and node are you using?)
npm: '6.14.15', node: '14.18.1'

**Comment by nadezdatsygankova** [comment]
> Great first pull request! 👍

thank you)))

File: backend/routes/login.js [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
-
-  // find that user in the database
-  const foundUser = await Principle.findOne({ email, password }).exec();
-
-  if (foundUser) {
+  console.log ("welcome to the login page");
+  try {
+    const principle = await Principle.findOne({email})
+    if(!principle){
+      return res.status(400).send({error:'Unable to login'})
```
[the remark this answers is not in the document]
Comment (nadezdatsygankova) [reply backend/routes/login.js:12]: agree, I will change it

[the remark this answers is not in the document]
Comment (nadezdatsygankova) [reply backend/routes/login.js:12]: > agree, I will change it

agree, I will change it

[the remark this answers is not in the document]
Comment (nadezdatsygankova) [reply backend/routes/login.js:12]: > This is the case when the user is not in our database. We can use an appropriate status code for this: [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404)
> 
> Also, It is better to specify the reason why user can't log in. So, something like: user not found / does not exist

agree, I will change it

File: backend/routes/login.js [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
+    }
+    const isMatch = await bcrypt.compare(password,principle.password )
+    if(!isMatch){
+      return res.status(400).send({error:'Unable to login'})
+    }
```
[the remark this answers is not in the document]
Comment (nadezdatsygankova) [reply backend/routes/login.js:17]: > We can use the [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401#:~:text=The%20HyperText%20Transfer%20Protocol%20(HTTP,credentials%20for%20the%20requested%20resource) status code for this case.
> 
> And we can describe the reason why user can't log in.

I will do it. Thank you

File: backend/routes/login.js [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
 
-    // send the user information to the front end including names , family info .. etc 
-    console.log("foundit");
-  } else {
-    // send error message to the front end 
-    console.log("check the username and password again");
+    res.send(principle)
```
[the remark this answers is not in the document]
Comment (nadezdatsygankova) [reply backend/routes/login.js:19]: > https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

I will change it

File: backend/package-lock.json (hunk 1 above)
Comment (nadezdatsygankova) [reply backend/package-lock.json:5]: I updated my npm version and lockfileVersion right now is 3

**Review by AikaAsan — APPROVED** [review]

**Comment by AikaAsan** [comment]
lgtm

**Comment by i-iurchuk** [comment]
Good job! +1

**Review by i-iurchuk — APPROVED** [review]

**Pull request merged**